hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/support/validators/block/StateChangesValidator.java

```diff
@@ -0,0 +1,728 @@
+import com.swirlds.state.State;
+import com.swirlds.state.spi.CommittableWritableStates;
+import com.swirlds.state.spi.Service;
+import com.swirlds.state.spi.info.NetworkInfo;
+import com.swirlds.state.spi.info.NodeInfo;
+import com.swirlds.state.spi.info.SelfNodeInfo;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.InstantSource;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+import java.util.stream.StreamSupport;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.Assertions;
+
+/**
+ * A validator that asserts the state changes in the block stream, when applied directly to a {@link MerkleStateRoot}
+ * initialized with the genesis {@link Service} schemas, result in the given root hash.
+ */
+public class StateChangesValidator implements BlockStreamValidator {
+    private static final Logger logger = LogManager.getLogger(StateChangesValidator.class);
+    private static final MerkleCryptography CRYPTO = MerkleCryptoFactory.getInstance();
+
+    private static final int HASH_SIZE = 48;
+    private static final int VISUALIZATION_HASH_DEPTH = 5;
+    private static final Pattern NUMBER_PATTERN = Pattern.compile("\\d+");
+    private static final Pattern STATE_ROOT_PATTERN = Pattern.compile(".*MerkleStateRoot.*/.*\\s+(.+)");
+    private static final Pattern CHILD_STATE_PATTERN = Pattern.compile("\\s+\\d+ \\w+\\s+(\\S+)\\s+.+\\s+(.+)");
+
+    private final Path pathToNode0SwirldsLog;
+    private final Bytes expectedRootHash;
+    private final Set<String> servicesWritten = new HashSet<>();
+    private final StateChangesSummary stateChangesSummary = new StateChangesSummary(new TreeMap<>());
+
+    private Timestamp genesisMigrationTimestamp = null;
+    private MerkleStateRoot state;
+
+    public static void main(String[] args) {
+        final var validator = new StateChangesValidator(
+                Bytes.fromHex(
+                        "a1d57e05d0607f76d13bc2ed31c40585adc294927e767df1f3443f065cc7a3348d3e0c4dd5581e6922c718e16b787231"),
+                Paths.get(
+                        "/Users/michaeltinker/AlsoDev/hedera-services/hedera-node/test-clients/build/hapi-test/node0/output/swirlds.log"),
+                Paths.get(
+                        "/Users/michaeltinker/AlsoDev/hedera-services/hedera-node/test-clients/build/hapi-test/node0/config.txt"));
+        final var input =
+                "/Users/michaeltinker/AlsoDev/hedera-services/hedera-node/test-clients/build/hapi-test/node0/data/block-streams/block-0.0.3/";
```

Comment:
Also, intentional?

Reply:
The main method is used for local testing.